## Rosalind.info
Hi, my name is Natalie!

This repository contains my personal solutions to the Rosalind.info bioinformatics problems. 

As I work through these problems, I both wanted to practice Python (with a bioinformatics focus) and archive my solutions.

I hope you find this informational and helpful!

*Also, please don't use these solutions without permission.*

## Progress
![Progress](https://progress-bar.dev/8/)
| ID | Title      | Solved? |
| ----------- | ----------- | ----------- |
|DNA|	Counting DNA Nucleotides|✓|
|RNA|	Transcribing DNA into RNA|✓|
|REVC|	Complementing a Strand of DNA|✓|
|GC	|Computing GC Content|✓|
|HAMM|	Counting Point Mutations|✓|
|PROT|	Translating RNA into Protein|✓|
|SUBS|	Finding a Motif in DNA|✓|
|PRTM|	Calculating Protein Mass|✓|
|FIB|	Rabbits and Recurrence Relations||
|IPRB|	Mendel's First Law||
|GRPH|	Overlap Graphs||
|LCSM|	Finding a Shared Motif||
|MPRT|	Finding a Protein Motif||
|MRNA|	Inferring mRNA from Protein||
|ORF|	Open Reading Frames||
|PERM|	Enumerating Gene Orders||
|REVP|	Locating Restriction Sites||
|SPLC|	RNA Splicing||
|CONS|	Consensus and Profile||
|FIBD|	Mortal Fibonacci Rabbits||
|IEV|	Calculating Expected Offspring||
|LIA|	Independent Alleles||
|LEXF|	Enumerating k-mers Lexicographically||
|LGIS|	Longest Increasing Subsequence||
|LONG|	Genome Assembly as Shortest Superstring||
|PMCH|	Perfect Matchings and RNA Secondary Structures||
|PPER|	Partial Permutations||
|PROB|	Introduction to Random Strings||
|SIGN|	Enumerating Oriented Gene Orderings||
|SSEQ|	Finding a Spliced Motif||
|TRAN|	Transitions and Transversions||
|TREE|	Completing a Tree||
|CAT|	Catalan Numbers and RNA Secondary Structures||
|CORR|	Error Correction in Reads||
|INOD|	Counting Phylogenetic Ancestors||
|KMER|	k-Mer Composition||
|KMP|	Speeding Up Motif Finding||
|LCSQ|	Finding a Shared Spliced Motif||
|LEXV|	Ordering Strings of Varying Length Lexicographically||
|MMCH|	Maximum Matchings and RNA Secondary Structures||
|PDST|	Creating a Distance Matrix||
|REAR|	Reversal Distance||
|RSTR|	Matching Random Motifs||
|SSET|	Counting Subsets||
|ASPC|	Introduction to Alternative Splicing||
|EDIT|	Edit Distance||
|EVAL	|Expected Number of Restriction Sites||
|MOTZ|	Motzkin Numbers and RNA Secondary Structures||
|NWCK|	Distances in Trees||
|SCSP|	Interleaving Two Motifs||
|SETO|	Introduction to Set Operations||
|SORT|	Sorting by Reversals||
|SPEC|	Inferring Protein from Spectrum||
|TRIE|	Introduction to Pattern Matching||
|CONV|	Comparing Spectra with the Spectral Convolution||
|CTBL|	Creating a Character Table||
|DBRU|	Constructing a De Bruijn Graph||
|EDTA|	Edit Distance Alignment||
|FULL	|Inferring Peptide from Full Spectrum||
|INDC	|Independent Segregation of Chromosomes||
|ITWV|	Finding Disjoint Motifs in a Gene||
|LREP|	Finding the Longest Multiple Repeat||
|NKEW|	Newick Format with Edge Weights||
|RNAS|	Wobble Bonding and RNA Secondary Structures||
|AFRQ|	Counting Disease Carriers||
|CSTR|	Creating a Character Table from Genetic Strings||
|CTEA|	Counting Optimal Alignments||
|CUNR|	Counting Unrooted Binary Trees||
|GLOB|	Global Alignment with Scoring Matrix||
|PCOV|	Genome Assembly with Perfect Coverage||
|PRSM|	Matching a Spectrum to a Protein||
|QRT|	Quartets||
|SGRA|	Using the Spectrum Graph to Infer Peptides||
|SUFF|	Encoding Suffix Trees||
|CHBP|	Character-Based Phylogeny||
|CNTQ|	Counting Quartets||
|EUBT|	Enumerating Unrooted Binary Trees||
|GASM|	Genome Assembly Using Reads||
|GCON|	Global Alignment with Constant Gap Penalty||
|LING|	Linguistic Complexity of a Genome||
|LOCA|	Local Alignment with Scoring Matrix||
|MEND|	Inferring Genotype from a Pedigree||
|MGAP|	Maximizing the Gap Symbols of an Optimal Alignment||
|MREP|	Identifying Maximal Repeats||
|MULT|	Multiple Alignment||
|PDPL|	Creating a Restriction Map||
|ROOT|	Counting Rooted Binary Trees||
|SEXL|	Sex-Linked Inheritance||
|SPTD|	Phylogeny Comparison with Split Distance||
|WFMD|	The Wright-Fisher Model of Genetic Drift||
|ALPH|	Alignment-Based Phylogeny||
|ASMQ|	Assessing Assembly Quality with N50 and N75||
|CSET|	Fixing an Inconsistent Character Set||
|EBIN|	Wright-Fisher's Expected Behavior||
|FOUN|	The Founder Effect and Genetic Drift||
|GAFF|	Global Alignment with Scoring Matrix and Affine Gap Penalty||
|GREP|	Genome Assembly with Perfect Coverage and Repeats||
|OAP|	Overlap Alignment||
|QRTD|	Quartet Distance||
|SIMS|	Finding a Motif with Modifications||
|SMGB|	Semiglobal Alignment||
|KSIM|	Finding All Similar Motifs||
|LAFF|	Local Alignment with Affine Gap Penalty||
|OSYM|	Isolating Symbols in Alignments||
|RSUB|	Identifying Reversing Substitutions||
